File: ch02/ch02_12.py
with open('col1.txt', 'w') as col1_f:
    with open('col2.txt', 'w') as col2_f:
      with open('popular-names.txt', 'r') as f:
          for line in f:
              col1, col2, *_ = line.split('\t')
              col1_f.write(col1 + '\n')
              col2_f.write(col2 + '\n')

File: ch02/ch02_12.rb
File.open('col1.txt', 'w') do |col1_f|
  File.open('col2.txt', 'w') do |col2_f|
    File.foreach('popular-names.txt') do |line|
      col1, col2, _ = line.split(/\t/)
      col1_f.puts(col1)
      col2_f.puts(col2)
    end
  end
end